docs/docs/static-vs-normal-queries.md

```diff
@@ -45,18 +31,97 @@ The redux `staticQueryComponents` is a map from component jsonName to StaticQuer
 }
 ```
 
-The `staticQueryComponents` redux namespace is owned by the `static-query-components.js` reducer with reacts to `REPLACE_STATIC_QUERY` actions.
+In the example above, `blog-2018-07-17-announcing-gatsby-preview-995` is the key, with the object as its value in the `Map`.
+
+The `staticQueryComponents` Redux namespace watches for updates to queries while you are developing, and adds new data to your cache when queries change.
+
+## Replacing queries with JSON imports
+
+With the final build, there isn't a GraphQL server running to query for data. Gatsby has already [extracted](/docs/query-extraction/) and [run](/docs/query-execution/) the queries, [storing](/docs/query-execution/#save-query-results-to-redux-and-disk) them in files based on hashes in `/public/static/d/<hash>.json`. It can now remove code for GraphQL queries, because there isn't a server it can query against and the data is already available.
 
-It is created in query-watcher. TODO: Check other usages
+### Distinguishing between static and normal queries
 
-TODO: in query-watcher.js/handleQuery, we remove jsonName from dataDependencies. How did it get there? Why is jsonName used here, but for other dependencies, it's a path?
+Babel traverses all of your source code looking for queries during query extraction, in order for Gatsby to handle static and normal queries differently, it looks for 3 specific cases in [`babel-plugin-remove-graphql-queries`](https://github.com/gatsbyjs/gatsby/blob/master/packages/babel-plugin-remove-graphql-queries/src/index.js):
 
-### Usages
+1. JSX nodes with the name `StaticQuery`
+2. Calls to functions with the name `useStaticQuery`
+3. Tagged template expressions using the `gql` tag
 
-- [websocket-manager](#TODO). TODO
-- [query-watcher](#TODO).
+### Adding imports for page data
 
-  - `getQueriesSnapshot` returns map with snapshot of `state.staticQueryComponents`
-  - handleComponentsWithRemovedQueries. For each staticQueryComponent, if passed in queries doesn't include `staticQueryComponent.componentPath`. TODO: Where is StaticQueryComponent created? TODO: Where is queries passed into `handleComponentsWithRemovedQueries`?
+Code specific for querying with GraphQL is no longer relevant and can be swapped out.
+
+A static query written in a component with the `useStaticQuery` hook like this:
```

Comment:
Not sure what the transition here is? Maybe a brief introduction about this being an example?

Reply:
```suggestion
The imports and query declarations are changed to imports for the JSON that correspond to the query result that Gatsby found when it ran the query. Consider the following component with with a static query written using the `useStaticQuery` hook:
```